iphone/cli/commands/_buildModule.js

```diff
@@ -0,0 +1,803 @@
+				if (bindingMatch != null) {
+					while (bindingMatch != null) {
+						keyList.push(bindingMatch[1]);
+						bindingMatch = bindingReg.exec(value);
+					}
+
+					keyList.forEach(function (key) {
+						if (this.tiXcconfig[key]) {
+							value = value.replace('$('+key+')', this.tiXcconfig[key]);
+						}
+					}, this);
+				}
+				this.tiXcconfig[match[1].trim()] = value;
+			}
+		}, this);
+	}
+
+	next();
+};
+
+iOSModuleBuilder.prototype.compileJS = function (next) {
+
+	this.jsFilesToEncrypt = [];
+
+	var moduleJS = this.manifest.moduleid + '.js',
+		jsFile = path.join(this.assetsDir, moduleJS),
+		renderData = {
+			'moduleIdAsIdentifier' : this.moduleIdAsIdentifier,
+			'mainEncryptedAssetReturn': 'return filterDataInRange([NSData dataWithBytesNoCopy:data length:sizeof(data) freeWhenDone:NO], ranges[0]);',
+			'allEncryptedAssetsReturn': 'NSNumber *index = [map objectForKey:path];'
+				+ '\n\t\tif (index == nil) {\n\t\t\treturn nil;\n\t\t}'
+				+ '\n\t\treturn filterDataInRange([NSData dataWithBytesNoCopy:data length:sizeof(data) freeWhenDone:NO], ranges[index.integerValue]);'
+		},
+		titaniumPrepHook = this.cli.createHook('build.ios.titaniumprep', this, function (exe, args, opts, done) {
+			var tries = 0,
+				completed = false,
+				jsFilesToEncrypt = opts.jsFiles,
+				placeHolderName = opts.placeHolder;
+
+			this.logger.info('Encrypting JavaScript files: %s', (exe + ' "' + args.join('" "') + '"').cyan);
+			jsFilesToEncrypt.forEach(function (file) {
+				this.logger.debug(__('Preparing %s', file.cyan));
+			}, this);
+
+			async.whilst(
+				function () {
+					if (tries > 3) {
+						// we failed 3 times, so just give up
+						this.logger.error(__('titanium_prep failed to complete successfully'));
+						this.logger.error(__('Try cleaning this project and build again') + '\n');
+						process.exit(1);
+					}
+					return !completed;
+				},
+				function (cb) {
+					var child = spawn(exe, args, opts),
+						out = '';
+
+					child.stdin.write(jsFilesToEncrypt.join('\n'));
+					child.stdin.end();
+
+					child.stdout.on('data', function (data) {
+						out += data.toString();
+					});
+
+					child.on('close', function (code) {
+						if (code) {
+							this.logger.error(__('titanium_prep failed to run (%s)', code) + '\n');
+							process.exit(1);
+						}
+
+						if (out.indexOf('initWithObjectsAndKeys') !== -1) {
+							// success!
+							renderData[placeHolderName] = out;
+
+							completed = true;
+						} else {
+							// failure, maybe it was a fluke, try again
+							this.logger.warn(__('titanium_prep failed to complete successfully, trying again'));
+							tries++;
+						}
+						cb();
+					}.bind(this));
+				}.bind(this),
+				done
+			);
+		});
+
+	var tasks = [
+		// 1. compile module js
+		function (cb) {
+			fs.existsSync(jsFile) && this.jsFilesToEncrypt.push(jsFile);
+
+			if (!this.jsFilesToEncrypt.length) {
+				renderData['mainEncryptedAsset'] = '';
```

Comment:
Throughout this section, you're using bracket notation rather than dotted notation. Is there a reason?